_posts/2021-08-03-scalable-ets-counters.md

```diff
@@ -0,0 +1,222 @@
+---
+layout: post
+title: Decentralized ETS Counters for Better Scalability
+tags: ETS, erlang term storage, scalability, multicore
+author: Kjell Winblad
+---
+
+A shared [Erlang Term Storage
+(ETS)](https://erlang.org/doc/man/ets.html) table is often an
+excellent place to store data that that is updated and read from
+multiple Erlang processes frequently. ETS provides key-value stores to
+Erlang processes. When the
+[write_concurrency](https://erlang.org/doc/man/ets.html#new-2) option
+is activated, ETS tables use fine-grained locking
+internally. Therefore, a scenario where multiple processes insert and
+remove different items in an ETS table should scale well with the
+number of utilized cores. Unfortunately, in practice, the scalability
+for such scenarios is not yet perfect. However, the ETS table option
+[`decentralized_counters`](https://erlang.org/doc/man/ets.html#new-2)
+(introduced in Erlang/OTP 22 for `ordered_set` tables and in
+Erlang/OTP 23 for the other table types) has made the scalability much
+better. A table with `decentralized_counters` activated uses
+decentralized counters instead of centralized counters to track the
+number of items in the table and the memory consumption. This blog
+post will give you a better understanding of when one should activate
+the `decentralized_counters` option and how the decentralized counters
+work.
+
```

Comment:
I think it would be useful to mention already here in the introduction what the disadvantage of the `decentralized_counters` option is. If found myself reading through the rest of blog post quickly until I found that information at the end.

Reply:
Fixed in b8fb1b3. Please have a look to see if you think it is better now.